### DISCUSSION TITLE
Set ENV Variable Using Github Action Workflow CMD

### DISCUSSION THREAD

**Discussion by varunsridharan** [community/community#25993] [Actions]
<p>Hi There</p>
<p>i am currently working on a simple github action for my personal use in which i need to set some ENV Variable on the go. (Fully Dynamic) and if its a simple string it works great when i use</p>
<pre><code class="lang-auto">::set-env name=${key}::${value}
</code></pre>
<p>but thats not the case here</p>
<p>i need to create below YAML Env content dynamically using PHP</p>
<pre><code class="lang-yaml">env:
  VAR1: |
    value1
    value2
</code></pre>
<p>in the above example you can see i have used <em><strong>|</strong></em> to enter multi line values.</p>
<p>As of now using <strong>PHP</strong> i was able to set variables if its a single line values</p>
<p><em><strong>Example</strong></em></p>
<pre><code class="lang-auto">::set-env name=VAR1::VALUE1
</code></pre>
<p>but if i try like this</p>
<pre><code class="lang-auto">::set-env name=VAR2::|
    value1
    value2
</code></pre>
<p>My other action is able to read only the <strong>|</strong>  not the full string :frowning:</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/varunsridharan">@varunsridharan</a>,</p>
<p>If you use the “<strong>set-env</strong>” command to set an environment variable with multi-line value, you need to escape the characters ‘<strong>%</strong>’, ‘<strong>\n</strong>’ and ‘<strong>\r</strong>’ in the value before passing it to the new environment variable. The runner will unescape in reverse.<br>
A simple demo as reference:</p>
<pre><code class="lang-yaml">- name: set env
  shell: bash
  run: |        
    multi_line="Line 01
    Line 02
    Line 03"

# escape the characters '%', '\n' and '\r'
    multi_line="${multi_line//'%'/'%25'}"
    multi_line="${multi_line//$'\n'/'%0A'}"
    multi_line="${multi_line//$'\r'/'%0D'}"

# pass the value to the new environment variable
    echo "::set-env name=MY_ENV::$multi_line"

- name: print env
  shell: bash
  run: echo "$MY_ENV"
</code></pre>
